osu.Framework/IO/AsyncBufferStream.cs

```diff
@@ -192,7 +192,7 @@ public override int Read(byte[] buffer, int offset, int count)
                 }
             }
 
-            Array.Copy(data, position, buffer, offset, amountBytesToRead);
+            data.AsSpan(position).CopyTo(buffer[..amountBytesToRead]);
```

Comment:
`amountBytesToRead` can be smaller than `data.Length - position`, which means that the destination span may become smaller than the source span, which is an [invalid operation](https://docs.microsoft.com/en-us/dotnet/api/system.span-1.copyto?view=net-6.0#exceptions).
```suggestion
            data.AsSpan(position, amountBytesToRead).CopyTo(buffer);
```

Reply:
You're right. I did not test this.